CenterForOpenScience/ember-osf-web
# [ENG-4963] Addon terms

-   Ticket: [ENG-4963]
-   Feature flag: n/a

## Purpose
- Add a Terms of Service component to show what each external service can do

## Summary of Changes
- Update External Service models to add a `termsOfService` attribute
- Update mirage fixtures to implement new `termsOfService` attribute
- Add new `AddonsService::TermsOfService` component and invoke it in the user settings page and node addons page
- Styling

## Screenshot(s)
- Gitlab ToS on node addons page
<img width="1124" alt="image" src="https://github.com/CenterForOpenScience/ember-osf-web/assets/51409893/75334608-8672-4b1d-8c98-4bfd82797af8">

- figshare ToS on node addons page
<img width="1127" alt="image" src="https://github.com/CenterForOpenScience/ember-osf-web/assets/51409893/81bed937-8d35-4436-90ec-305ccbb4cd32">

- Box ToS on user settings page
<img width="1305" alt="image" src="https://github.com/CenterForOpenScience/ember-osf-web/assets/51409893/dbf7958f-e082-4cbc-87e9-dc7569c37174">

- Zotero ToS on the user settings page (mobile)
<img width="277" alt="image" src="https://github.com/CenterForOpenScience/ember-osf-web/assets/51409893/c1bd0bd5-90ad-410d-a12f-395ef9aa10fa">

- Boa ToS on the node addons page (mobile)
<img width="303" alt="image" src="https://github.com/CenterForOpenScience/ember-osf-web/assets/51409893/c5ffd29c-6987-4395-8b80-80544d544bfb">



## Side Effects
- there's some gross TypeScript in there...

## QA Notes
- Double-check capabilities and language based on Addon Permission Language spreadsheet

[ENG-4963]: https://openscience.atlassian.net/browse/ENG-4963?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
